File: back-end/src/dto/player.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class PlayerState(BaseModel):
    owner_id: UUID
    active_playlist_id: UUID | None = None
    current_track_id: str | None = None
    current_track_data: dict[str, Any] | None = None
    position: float = 0.0
    is_paused: bool = True
    volume: int = 80
    broadcast_to_widget: bool = True
    last_client_id: str | None = None
    updated_at: datetime | str | None = None

    model_config = ConfigDict(from_attributes=True)


class PlayerPlayRequest(BaseModel):
    track_id: str
    playlist_id: UUID
    client_id: str
    position: float = 0.0


class PlayerPauseRequest(BaseModel):
    is_paused: bool
    position: float = 0.0
    client_id: str


class PlayerSeekRequest(BaseModel):
    position: float
    client_id: str


class PlayerVolumeRequest(BaseModel):
    volume: int = Field(..., ge=0, le=100)
    client_id: str


class PlayerBroadcastRequest(BaseModel):
    enabled: bool
    client_id: str

File: new_ui/src/types/player.ts
export type PlayerMode = 'listen' | 'control'

export interface PlayerState {
  owner_id: string
  active_playlist_id: string | null
  current_track_id: string | null
  current_track_data: Record<string, any> | null
  position: number
  is_paused: boolean
  volume: number
  broadcast_to_widget: boolean
  last_client_id: string | null
  updated_at?: string | null
}

export interface PlayerPlayRequest {
  track_id: string
  playlist_id: string
  client_id: string
  position?: number
}

export interface PlayerPauseRequest {
  is_paused: boolean
  position: number
  client_id: string
}

export interface PlayerSeekRequest {
  position: number
  client_id: string
}

export interface PlayerVolumeRequest {
  volume: number
  client_id: string
}

export interface PlayerBroadcastRequest {
  enabled: boolean
  client_id: string
}

export interface ActiveChannel {
  owner_id: string
  name: string
  is_owner: boolean
  can_control_player: boolean
  can_manage_all_playlists: boolean
}
